security/cmd/node_agent/main.go

```diff
@@ -57,11 +80,18 @@ func init() {
 		"The requested TTL for the workload")
 	flags.IntVar(&cAClientConfig.RSAKeySize, "key-size", 2048, "Size of generated private key")
 	flags.StringVar(&cAClientConfig.CAAddress,
-		"ca-address", "istio-citadel:8060", "Istio CA address")
-
-	flags.StringVar(&cAClientConfig.Env, "env", "unspecified",
+		"ca-address", os.Getenv(caAddress), "Istio CA address")
+	flags.StringVar(&cAClientConfig.CAProviderName, "CAProviderName", os.Getenv(caProvider), "Name of the CA")
+	userCaProtocol := oldCAProtocolIndicator
+	flags.StringVar(&userCaProtocol, "CAProtocol", os.Getenv(caProtocol), "CA protocol")
```

Comment:
Use `env.RegisterStringVar` such that documentation can be auto generated.  I saw pilot use this lib a lot.
https://github.com/istio/istio/blob/master/pkg/env/var.go

@geeknoid i think that's recommended way, right? similar to what we did for automation?

Reply:
Also why add flags? We don't need to add flag to propogate from env var. That still allows people to play with flags.

I believe the node_agent_k8s use flags is just for migration since it's using flags before. This is new config option, no need to add flag.

Reply:
also move to constants package.